src/opencast.js

```diff
@@ -86,9 +90,47 @@ export class Opencast {
 
     await catchRequestError(async () => await self.updateUser());
 
+    await self.handleSeriesIDSelection();
+
     return self;
   }
 
+
+  async getSeriesID() {
+    let headers = this.header();
+    let url = `${this.getServerUrl()}/admin-ng/series/series.json`
+    await fetch(url, {
+      credentials: 'same-origin',
+      redirect: 'manual',
+      headers,
+      method: 'get'
+    }).then(response => response.json()).then(data => {
+      for (let root in data) {
+        for (let enumerator in data[root]) {
+          this.IDS.set(data[root][enumerator]['id'], data[root][enumerator]['title']);
+        }
+      }
+    }).catch(error => this.IDS = {});
+
+  }
+
+
+
+  header() {
+    let headers = {};
+    if (this.getLoginUsername() && this.getLoginPassword()) {
+      const encoded = btoa(unescape(encodeURIComponent(
+          this.getLoginUsername() + ':' + this.getLoginPassword()
+      )));
+      headers = {'Authorization': `Basic ${encoded}`};
+    }
+    return headers;
+  }
+
+  async handleSeriesIDSelection() {
+    await this.getSeriesID();
+  }
```

Comment:
Why this extra wrapper method?

Reply:
This is basically a relic from an earlier implementation, that i forgot to remove. I will put the function content from getSeriesID() in there and remove the function.